src/shared/lib/extendCBioPortalAPI.ts

```diff
@@ -0,0 +1,113 @@
+import { SiteError } from 'AppStore';
+import { getBrowserWindow } from 'cbioportal-frontend-commons';
+import { Simulate } from 'react-dom/test-utils';
+import error = Simulate.error;
+import autobind from 'autobind-decorator';
+import request from 'superagent';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+export function extendCBioPortalAPI<T extends Constructor<{}>>(Base: T) {
```

Comment:
extend is kind of vague and since this is framework magic I think it's important to make the name more specific if possible. maybe like `addErrorHandlingToAPI`?

Reply:
Should we also add some comments here to explain the details? This function seems pretty complex to me.